File: wscodeblocks/hfsguardadir-foxtoolkit/hfsguardadir/objetos/Aba.cpp
#include "Aba.h"

namespace hfsguardadir {

namespace objetos {

Aba::Aba()
{
    this->limparDados();
}

Aba::Aba(int codigo, const std::string &nome, int ordem) {
    this->codigo = codigo;
    this->nome = nome;
    this->ordem = ordem;
}

int Aba::getCodigo() {
    return codigo;
}

void Aba::setCodigo(int codigo) {
    this->codigo = codigo;
}

std::string &Aba::getNome() {
    return nome;
}

void Aba::setNome(const std::string &nome) {
    this->nome = nome;
}

int Aba::getOrdem() {
    return ordem;
}

void Aba::setOrdem(int ordem) {
    this->ordem = ordem;
}

void Aba::limparDados() {
    this->codigo = 0;
    this->nome = "";
    this->ordem = 0;
}

bool Aba::equals(Aba obj) {
    return (obj.getCodigo() == this->getCodigo()
        && obj.getNome() == this->getNome()
        && obj.getOrdem() == this->getOrdem());
}

std::string &Aba::toString() {
    return this->nome;
}

}

}

File: HFSGuardaDiretorio_Java/hfsguardadirSWT/src/main/java/hfsguardadir/objetos/Aba.java
package hfsguardadir.objetos;

import hfsguardadir.comum.Rotinas;

public final class Aba {

    private int codigo;
    private String nome;
    private int ordem;

    public Aba() {
        this.limparDados();
    }

    public Aba(int codigo, String nome, int ordem) {
        this.codigo = codigo;
        this.nome = nome;
        this.ordem = ordem;
    }

    public int getCodigo() {
        return codigo;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getOrdem() {
        return ordem;
    }

    public void setOrdem(int ordem) {
        this.ordem = ordem;
    }

    public void limparDados() {
        this.codigo = 0;
        this.nome = "";
        this.ordem = 0;
    }

    public boolean equals(Aba obj) {
        return (obj.getCodigo() == this.getCodigo()
                && obj.getNome().equals(this.getNome())
                && obj.getOrdem() == this.getOrdem());
    }

    @Override
    public String toString() {
        return this.nome;
    }

    public String toString2() {
        return Rotinas.getRecurso("Aba.codigo") + " " + this.codigo + ", "
                + Rotinas.getRecurso("Aba.nome") + " " + this.nome;
    }

}
